### DISCUSSION TITLE
Yolov8-seg and Yolov10 in one project

### DISCUSSION THREAD

**Discussion by JelluSandro** [ultralytics/ultralytics#14486] [Q&A]
Hello! In my project I need to use detection and segmentation both. For these tasks, I choose Yolov10 for detection and Yolov8 for segmentation. I successfully train both models. But I have a problem. I don't know hot to use them together in one project. I work in google colab and code looks like this:
```
!pip install -q git+https://github.com/THU-MIG/yolov10.git
!pip install ultralytics

from ultralytics import YOLO
from ultralytics import YOLOv10

model_segmentation = YOLO('/content/drive/MyDrive/tram_weights/tram_segmentation.pt')
model_detection = YOLOv10('/content/drive/MyDrive/datasets/tram_weights/detection.pt')

model_detection.predict(batch_imgs)
model_segmentation.predict(batch_imgs)
```

And detection work correctly as I expected. But segmentation got error
```
[<ipython-input-21-5bdf2ab82a71>](https://localhost:8080/#) in get_all_detection_by_batch(batch_images, model, dict_names, flag)
     29                 verbose=False)
     30         else:
---> 31             results = model.segment(batch_images,
     32                 show=False,
     33                 save=False,

[/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py](https://localhost:8080/#) in __getattr__(self, name)
   1707             if name in modules:
   1708                 return modules[name]
-> 1709         raise AttributeError(f"'{type(self).__name__}' object has no attribute '{name}'")
   1710 
   1711     def __setattr__(self, name: str, value: Union[Tensor, 'Module']) -> None:

AttributeError: 'YOLO' object has no attribute 'segment'
```

I think it because segmentation, try using lib for Yolov10. Is exists a correct way to install both? Or how I can use segmentation and detection both? May I train segmentation on custom dataset on Yolov10?

**Comment by pderrenger** [comment]
@JelluSandro hello! It looks like you're encountering an issue because the segmentation model is trying to use the YOLOv10 library, which doesn't support segmentation. To resolve this, ensure that you are using the correct methods from the Ultralytics YOLO library for segmentation. You should use `model.predict()` for both detection and segmentation tasks. Here's a quick tip: make sure you call the correct method for each model type. If the issue persists, please verify that you are using the latest versions of both libraries. Unfortunately, YOLOv10 does not support segmentation, so you'll need to stick with YOLOv8 for that task. If you need further assistance, feel free to ask!
